CMakeLists.txt

```diff
@@ -1,73 +1,71 @@
 
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_BUGFIX VER_BUILD VER_STATUS)
-# message(STATUS "Project version is: ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD} (${VER_STATUS})")
-
-project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
+project(qBittorrent VERSION ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD})
 
 # check for invalid compiler version/CXX standard as early as possible
+# ideally, we would also be able to check whether or not libtorrent was compiled
+# with the same standard from imported target properties
 include(FunctionQbtCXXCompilerAndModeCheck)
 qbt_minimum_cxx_mode_check(14)
-message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
-            "Make sure libtorrent was built with the same C++ mode for ABI compatibility.")
-
-set(PROJECT_VERSION "${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}")
-
-if (NOT VER_BUILD EQUAL 0)
-    set(PROJECT_VERSION "${PROJECT_VERSION}.${VER_BUILD}")
-endif()
-
-set(PROJECT_VERSION "${PROJECT_VERSION}${VER_STATUS}")
-
-add_definitions(-DQBT_VERSION_MAJOR=${VER_MAJOR})
-add_definitions(-DQBT_VERSION_MINOR=${VER_MINOR})
-add_definitions(-DQBT_VERSION_BUGFIX=${VER_BUGFIX})
-add_definitions(-DQBT_VERSION_BUILD=${VER_BUILD})
-
-add_definitions(-DQBT_VERSION="v${PROJECT_VERSION}")
-add_definitions(-DQBT_VERSION_2="${PROJECT_VERSION}")
-
-include(GNUInstallDirs)
-include(FeatureSummary)
 
 # version requirements
-set(requiredBoostVersion 1.40)
+set(requiredBoostVersion 1.60)
 set(requiredQtVersion 5.9.0)
-set(requiredOpensslVersion 1.0)
-set(requiredLibtorrentVersion 1.1.10)
+set(requiredOpenSSLVersion 1.0)
+set(requiredLibtorrentVersion 1.2.0)
+set(requiredZlibVersion 1.2.5.2)
 
-if (WIN32)
-    include(winconf)
+# features (some are platform-specific)
+include(FeatureSummary)
+include(FeatureOptionsSetup)
+feature_option(COUNTRIES_RESOLUTION "Enable resolving peers IP addresses to countries" ON)
+feature_option(STACKTRACE "Enable stacktraces" ON)
+feature_option(GUI "Build GUI application" ON)
+feature_option(WEBUI "Enables built-in HTTP server for headless use" ON)
+feature_option(VERBOSE_CONFIGURE "Show information about PACKAGES_FOUND and PACKAGES_NOT_FOUND in the configure output (only useful for debugging the CMake build scripts)" OFF)
+if (UNIX)
+    feature_option_dependent(DBUS
+        "Enables features such as notification support and optional power-management features on Linux via D-Bus"
+        ON "GUI" OFF
+    )
+    feature_option_dependent(POWER_MANAGEMENT
+        "Enables system power management features, such as preventing suspension while active"
+        ON "DBUS" OFF
+    )
+    feature_option_dependent(SYSTEMD
+        "Install systemd service file to a directory manually overridable with Systemd_SERVICES_INSTALL_DIR"
+        OFF "NOT GUI" OFF
+    )
+elseif (MSVC)
```

Comment:
Seems it should be another `if` block.

Reply:
I made the same suggestion earlier

Reply:
@glassez @NotTsunami 
I did it this way initially because I thought `if/elseif` makes more sense than `if/if` since these are mutually exclusive and related to the same thing (the features available on different platforms).